pylint/config.py

```diff
@@ -220,229 +285,30 @@ def _validate(value, optdict, name=''):
 def _level_options(group, outputlevel):
     return [option for option in group.option_list
             if (getattr(option, 'level', 0) or 0) <= outputlevel
-            and option.help is not optparse.SUPPRESS_HELP]
-
-
-def _expand_default(self, option):
-    """Patch OptionParser.expand_default with custom behaviour
-
-    This will handle defaults to avoid overriding values in the
-    configuration file.
-    """
-    if self.parser is None or not self.default_tag:
-        return option.help
-    optname = option._long_opts[0][2:]
-    try:
-        provider = self.parser.options_manager._all_options[optname]
-    except KeyError:
-        value = None
-    else:
-        optdict = provider.get_option_def(optname)
-        optname = provider.option_attrname(optname, optdict)
-        value = getattr(provider.config, optname, optdict)
-        value = utils._format_option_value(optdict, value)
-    if value is optparse.NO_DEFAULT or not value:
-        value = self.NO_DEFAULT_VALUE
-    return option.help.replace(self.default_tag, str(value))
-
-
-@contextlib.contextmanager
-def _patch_optparse():
-    orig_default = optparse.HelpFormatter
-    try:
-        optparse.HelpFormatter.expand_default = _expand_default
-        yield
-    finally:
-        optparse.HelpFormatter.expand_default = orig_default
+            and option.help is not argparse.SUPPRESS]
```

Comment:
(commenting here because I don't have other useful context) Won't we support any longer the various option kinds that we have now ``yn``, ``regexp_csv``, ``multiple_choices``? Or the plan is to add support for them after this PR? Sorry if you mentioned this already, I can't seem to find any reference to these in this PR.

Reply:
We can still support these. In `_convert_definition` I'm converting any existing validators from a string (eg `"yn"`, `"regexp_csv"`) to their validator functions so that argparse can use it.